lib/std/zig/AstGen.zig

```diff
@@ -8609,36 +8611,27 @@ fn numberLiteral(gz: *GenZir, ri: ResultInfo, node: Ast.Node.Index, source_node:
     }
 }
 
-fn failWithNumberError(astgen: *AstGen, err: std.zig.number_literal.Error, token: Ast.TokenIndex, bytes: []const u8) InnerError {
+fn failWithNumberError(
```

Comment:
There was a TODO master to DRY these string/error functions with a duplicate in Manifest.zig.

I also needed them, both in runtime and compile time ZON parsing, so rather than duplicate it further I resolved the TODO.

One awkward part of my resolution is that right now notes only get appended in AstGen.